### ISSUE TITLE
output_dir and yaml for lib directories vs style files.

### ISSUE THREAD

**Issue by jrounds** [rstudio/rmarkdown#474]
I posted an issue yesterday, this one is different.  This is a minor issue about lib_dir vs css fields in the yaml.

Now suppose my Rmarkdown document resides in /tmp, but ultimately I want to use lib and style directories in /tmp/01, and render an html into /tmp/01.  The file and directory structure prior to generating a report are this:

```
/tmp/foo.Rmd
/tmp/01/lib
/tmp/01/style.css
```

The appropriate yaml to get everything working right  (and as far as I know the only yaml) is:
```
---
output:
  html_document:
    self_contained: false
    lib_dir: 01/lib
    css: style.css
---
```

After a call to render("foo.Rmd", output_dir=/tmp/01),  the generated html files will have this in it:
```
<script src="lib/jquery-1.11.0/jquery.min.js"></script>
<link rel="stylesheet" href="style.css" type="text/css" />
```

A more intuitive yaml to me was this because it did not treat lib_dir and css as different relative to the generating .Rmd.
```
---
output:
  html_document:
    self_contained: false
    lib_dir: 01/lib
    css: 01/style.css
---
```
(In actual fact a more intuitive yaml to me was one without 01 because I wanted to think of the files as relative to the html but that isn't how it works.)

But if you do that you end up with html like this:
```
<script src="lib/jquery-1.11.0/jquery.min.js"></script>
<link rel="stylesheet" href="01/style.css" type="text/css" />
```

The issue is that 01 is the directory the html file is in, and that style sheet won't be found.  The point here is the css and lib_dir field have two different relative behaviors when used with output_dir.  I am just letting you know about this counter-intuitive aspect of yamls and rmarkdown render with output_dir that could slow down adoption.

**Comment by jrounds** [comment]
I am sorry to report that the logic of yaml's lib_dir and render's output_dir is just wonky, and you all need to set up a test case for it.

I ran into performance issues because I am working out of a VM that is having issues with a shared directory, and I thought, "I will just move the VMs markdown output to a tmp directory within the VM and let the intermediates directory be a ram disk."

But this is what I ran into:
Error in relativeTo(basepath, dir) : 
  The path /media/sf_work/15spring/weake_aging/04_03_edgeR/go_html_reports/lib/jquery-1.11.0 does not appear to be a descendant of /tmp/go_html_reports/

As far as I can tell there is *no way* to use a yaml with a lib_dir *and* use render's output_dir in an unrelated location.  There is as far as I can tell no way to do something like
source = /home/jrounds/proj1/report.Rmd

and have the output_dir argument of render be like:
output_dir = /tmp/report_html

You can put lib_dir where ever you want and do whatever with it in the yaml and render will choke on it.

You can do a directory structure like
```
/home/jrounds/proj1/report.Rmd
/home/jrounds/proj1/lib
/tmp/report_html/lib

with yaml however you think it should have lib_dir, and either way render("/home/jrounds/proj1/report.Rmd", output_dir = "/tmp/report_html") and render will error.

**Comment by jjallaire** [comment]
Yes it is definitely wonky. It grew up serving multiple use-cases and so is
not fully coherent and consistent. Cleaning this up is a priority but not
something we can make quick work of.

**Comment by jrounds** [comment]
I deleted the wonky post because I had included an intermediaries_dir and then it got really strange, and it wasn't because of output_dir: it was because of intermediaries_dir, and while there is something wonky going on with intermediaries_dir and yamls (imo) I didn't have time to parse out exactly what happened. I ended up just doing a quick copy of stuff to a temporary directory and muddling through that way.

Closing this because you all know of it.
